File: leetcode/2050-count-good-numbers/solution.c
const long long k = 1000000007;
long long fastpow(long long x, long long n) {
    if(n==0) return 1;
    long long p = fastpow(x, n/2);
    x = x*(n%2) + (n+1)%2;
    return p*p*x%k;
}
int countGoodNumbers(long long n) {
    return fastpow(4, n/2)*fastpow(5, n-n/2)%k;
}

File: leetcode/2050-count-good-numbers/solution.rs
impl Solution {
    pub fn count_good_numbers(n: i64) -> i32 {
        let k = 1000_000_007;
        let pow = |mut x, mut n| {
            let mut ret = 1;
            while n > 0 {
                if n%2 == 1 {
                    ret = ret*x%k;
                }
                x = x*x%k;
                n /= 2;
            }
            ret
        };
        (pow(4i64, n/2) * pow(5i64, n-n/2) % k) as i32
    }
}
